app/src/main/java/com/sun/basic_japanese/base/NavigationManager.kt

```diff
@@ -0,0 +1,68 @@
+package com.sun.basic_japanese.base
+
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentManager
+import com.sun.basic_japanese.R
+
+class NavigationManager(
+    private val fragmentManager: FragmentManager,
+    private val container: Int
+) {
+
+    val isRootFragmentVisible: Boolean
+        get() = fragmentManager.backStackEntryCount <= STACK_ENTRY_COUNT_LIMIT
+
+    private var navigationListener: (() -> Unit)? = null
+
+    init {
+        fragmentManager.addOnBackStackChangedListener {
+            navigationListener?.invoke()
+        }
+    }
+
+    fun open(fragment: Fragment) {
+        openFragment(fragment, addToBackStack = true, isRoot = false)
+    }
+
+    fun openAsRoot(fragment: Fragment) {
+        popEveryFragment()
+        openFragment(fragment, addToBackStack = false, isRoot = true)
+    }
+
+    fun navigateBack(): Boolean {
```

Comment:
Sử dụng dấu = nếu hàm chỉ return cho 1 giá trị 
```suggestion
    fun navigateBack(): Boolean = something 
```